src/main/scala/org/deepdive/ddlog/DeepDiveLogCompiler.scala

```diff
@@ -600,10 +676,10 @@ object DeepDiveLogCompiler extends DeepDiveLogHandler {
           weight = stmt.weights match {
             case KnownFactorWeight(x) => s"${x}"
             case UnknownFactorWeight(w) => {
-              val weightVar = w.flatMap(s => ss.resolveColumn(s, fakeCQ, AliasOnly)).mkString(", ")
-              if (weightVar == "") "?"
-              else s"?(${weightVar})"
+              val weightVar = w.map(s => qc.compileAlias(VarExpr(s), WeightAlias)).mkString(", ")
```

Comment:
We should perhaps change the parser to parse a whole expr for `UnknownFactorWeight` to support variables as well as constant literal strings?  Can we use a column holding arbitrary computed values for the weight ids in DeepDive btw?